Sources/AMQPClient/ChannelHandlers/AMQPChannelHandler.swift

```diff
@@ -11,421 +11,214 @@
-                default:
-                    preconditionUnexpectedPayload(payload)  
-                }
-            default:
-                preconditionUnexpectedPayload(payload)
-            }
-        case .header(let header):
-            self.nextMessage?.properties = header.properties
-        case .body(let body):
-            guard let msg = nextMessage, let properties = msg.properties else {
-                if let promise = self.responseQueue.popFirst() {
-                    promise.fail(AMQPConnectionError.invalidMessage)
-                }
-                return
-            }
-
-            switch msg.frame {
-            case .getOk(let getOk):
-                if let promise = self.responseQueue.popFirst() {
-                        promise.succeed(.channel(.message(.get(.init(
-                            message: .init(
-                                exchange: getOk.exchange,
-                                routingKey: getOk.routingKey,
-                                deliveryTag: getOk.deliveryTag,
-                                properties: properties,
-                                redelivered: getOk.redelivered,
-                                body: body),
-                            messageCount: getOk.messageCount)))))
-                }
-            case .deliver(let deliver):
-                self.notify(type: Delivery.self,
-                            named: deliver.consumerTag,
-                            .success(.init(
-                                exchange: deliver.exchange,
-                                routingKey: deliver.routingKey,
-                                deliveryTag: deliver.deliveryTag,
-                                properties: properties,
-                                redelivered: deliver.redelivered,
-                                body: body)))
-            case .return(let `return`):
-                self.notify(type: Return.self,
-                            .success(.init(
-                                replyCode: `return`.replyCode,
-                                replyText: `return`.replyText,
-                                exchange: `return`.exchange,
-                                routingKey: `return`.routingKey,
-                                properties: properties,
-                                body: body)))
-            default:
-                preconditionUnexpectedPayload(payload)
-            }
-
-            nextMessage = nil
-        default:
-            preconditionUnexpectedPayload(payload)
+    func handleCancellation(consumerTag: String) {
+        if let listener = listeners.withLockedValue({ $0.consumeListeners[consumerTag] }) {
+            listener(.failure(AMQPConnectionError.consumerCancelled))
         }
+        removeListener(type: Delivery.self, named: consumerTag)
     }
 
-    func close(error: Error? = nil) {
-        let shouldClose = self.state.withLockedValue { state in
+    func reportAsClosed(error: Error? = nil) {
```

Comment:
close is otherwise used to "initiate a close" or to actually "close the channel" - this function is kind of the result of a closing. again, to me this is connected to a bigger story of naming scheme/semantic consistency that is not yet fully established.